Automattic/jetpack
# Add/waf actions

Introduces new hook actions to the WAF. As the `actions.php` file grew, it also became obvious that the WAF would benefit from a better organization of package logic. To that end, the functionality was added to the runner class.

Two new CLI commands have been added; the first sets up a jetpack site for the WAF, the second simply generates a new rules.php file.




#### Changes proposed in this Pull Request:

* A Jetpack activation hook has been added
* A check for upgrade has been added to the current `plugin_loaded` hook
* Class names have been switched to Snake_Case
* Core package logic has been moved into the `Waf_Runner` class

#### Jetpack product discussion



#### Does this pull request change what data or activity we track or use?


No

#### Testing instructions:





**TEST 1:** Test sites without a provisioned WAF aren't affected. Assumes the site is a fresh Docker site.
1. Checkout this branch
2. Open the localhost plugin's page.
3. Deactivate the Jetpack plugin
4. Delete the rules.php file if exists
5. Activate the Jetpack plugin

**Test 1 Expectation:** No rules.php file should be created (the WAF has not been provisioned yet)

**Test 2:** Test the CLI setup

1. From the docker CLI run `wp jetpack-waf setup normal`
 
**Test 2 Expectation:** A rules.php file will be created containing only `<?php`

**Test 3:** Test CLI generating rules

1. Delete the rules.php file
2. From the docker CLI run `wp jetpack-waf generate_rules`

**Test 3 Expectation:** A rules.php file will be created containing only `<?php`

**Test 4:** Test rule generation when activating Jetpack plugin

1. Delete the rules.php file
2. Open the localhost plugin's page
3. Deactivate the Jetpack plugin
4. Activate the Jetpack plugin

**Test 4 Expectation:** A rules.php file will be created containing only `<?php`

#### Other notes

* Class name changes are in line with other Jetpack package patterns
* Other Unit Tests may be warranted, but ran into some difficulties. @ArSn if you would like to link up to discuss, I'd be happy to!
